tso-server/src/main/java/org/apache/omid/tso/PersistenceProcessorImpl.java

```diff
@@ -204,15 +204,13 @@ private void maybeFlushBatch() {
     synchronized private void flush() {
         lastFlush.set(System.nanoTime());
 
-        boolean areWeStillMaster = true;
         if (!leaseManager.stillInLeasePeriod()) {
-            // The master TSO replica has changed, so we must inform the
-            // clients about it when sending the replies and avoid flushing
-            // the current batch of TXs
-            areWeStillMaster = false;
-            // We need also to clear the data in the buffer
-            writer.clearWriteBuffer();
-            LOG.trace("Replica {} lost mastership before flushing data", tsoHostAndPort);
+            if (batch.getNumEvents() > 0) {
+                // The master TSO replica has changed, so we must commit suicide
+                // We need also to clear the data in the buffer
+                writer.clearWriteBuffer();
+                panicker.panic("Replica " + tsoHostAndPort + " lost mastership before flushing data. Committing suicide");
```

Comment:
super minor: 
do you want to move this to a method and replace two identical code with one method call? 

`panicker.panic("Replica " + tsoHostAndPort + " lost mastership before flushing data. Committing suicide");`